3-networks/main.tf

```diff
@@ -38,61 +51,81 @@ data "google_projects" "prod_host_project" {
+  environment_code     = local.nonprod_environment_code
+  vpc_label            = local.private_vpc_label
+  nat_region           = var.nat_region
+  private_service_cidr = local.nonprod_cidrs_private_service
+  bgp_asn_nat          = "64512"
   subnets = [
     {
-      subnet_name           = "example-subnet"
-      subnet_ip             = "10.200.4.0/22"
-      subnet_region         = var.default_region
+      subnet_ip             = local.nonprod_cidrs_sub_network1
+      subnet_region         = var.subnet_region1
       subnet_private_access = "true"
       subnet_flow_logs      = "false"
+      bgp_asn               = [64514, 64515]
       description           = "Non prod example subnet."
+      secondary_ranges = [
+        {
+          range_label   = "gke-pod"
+          ip_cidr_range = "192.168.0.0/19"
+        },
+        {
+          range_label   = "gke-svc"
+          ip_cidr_range = "192.168.32.0/23"
+        }
+      ]
     },
+    {
+      subnet_ip             = local.nonprod_cidrs_sub_network2
+      subnet_region         = var.subnet_region2
+      subnet_private_access = "true"
+      subnet_flow_logs      = "false"
+      bgp_asn               = [64516, 64517]
+      description           = "Non prod example subnet."
+      secondary_ranges      = []
+    }
   ]
-  secondary_ranges = {
-    example-subnet = [
-      {
-        range_name    = "example-subnet-gke-pod"
-        ip_cidr_range = "192.168.0.0/19"
-      },
-      {
-        range_name    = "example-subnet-gke-svc"
-        ip_cidr_range = "192.168.32.0/23"
-      },
-    ]
-  }
 }
 
 module "shared_vpc_prod" {
   source               = "./modules/standard_shared_vpc"
   project_id           = local.prod_host_project_id
-  default_region       = var.default_region
-  network_name         = "shared-vpc-prod"
-  private_service_cidr = "10.20.0.0/22"
-  bgp_asn              = 64513
+  environment_code     = local.prod_environment_code
+  vpc_label            = local.private_vpc_label
+  nat_region           = var.nat_region
+  private_service_cidr = local.prod_cidrs_private_service
+  bgp_asn_nat          = 64513
   subnets = [
     {
-      subnet_name           = "example-subnet"
-      subnet_ip             = "10.20.20.0/22"
-      subnet_region         = var.default_region
+      subnet_ip             = local.prod_cidrs_sub_network1
+      subnet_region         = var.subnet_region1
       subnet_private_access = "true"
       subnet_flow_logs      = "false"
       description           = "Prod example subnet."
+      subnet_private_access = "true"
+      subnet_flow_logs      = "false"
+      bgp_asn               = [64518, 64519]
+      secondary_ranges = [
+        {
+          range_label   = "gke-pod"
+          ip_cidr_range = "192.168.96.0/19"
+        },
+        {
+          range_label   = "gke-svc"
+          ip_cidr_range = "192.168.128.0/23"
+        }
+      ]
     },
+    {
+      subnet_ip             = local.prod_cidrs_sub_network2
```

Comment:
As mentioned above, I think the CIDR block should be here - otherwise it hurts readability